### PR TITLE
Add tests for deployment manager resources and move autogenSpec to KRM resource

### PR DESCRIPTION
**Pull request by gibbleyg**

### CODE REVIEW

File: mpdev/internal/apply/deployment_manager_test.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+		{
+			name: "Deployment Manager Local Save Absolute Path",
+			expectedRunArgs: [][]string{
+				{"zip", "-r", "/tmp/dir3/localzippath.zip", "."},
+			},
+			zipFilePath: "/tmp/dir3/localzippath.zip",
+		},
+		{
+			name:            "Deployment Manager Missing Reference",
+			expectedRunArgs: nil,
```
Comment (huyhg) [inline mpdev/internal/apply/deployment_manager_test.go:63]: this this explicitness required? Should be `nil` if not specified, right?

File: mpdev/internal/apply/deployment_manager.go [hunk 2]
```diff
@@ -23,13 +23,14 @@ import (
...
 	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
 )
 
 // DeploymentManagerAutogenTemplate generates a deployment manager template
 // given an autogen.yaml file.
 type DeploymentManagerAutogenTemplate struct {
 	BaseResource
-	AutogenFile string
+	AutogenSpec interface{}
```
Comment (huyhg) [inline mpdev/internal/apply/deployment_manager.go:33]: Can you update the PR description as well?

Comment (gibbleyg) [reply mpdev/internal/apply/deployment_manager.go:33]: Done

File: mpdev/internal/apply/deployment_manager_test.go (hunk 1 above)
Comment (gibbleyg) [reply mpdev/internal/apply/deployment_manager_test.go]: Fixed

**Pull request merged**